projects/dynamicforms/src/components/upload/upload.component.ts

```diff
@@ -61,7 +61,6 @@ export class UploadComponent extends AbstractFormComponent implements OnInit {
     ) {
       this.fileList = Array.from(filteredFiles);
       this.fileList.splice(this.config.maxUploads);
-      this.setValueAndValidate(this.fileList);
```

Comment:
There is no need to set the control until user click on the UPLOAD button.

Reply:
Yes there is. Without it user won't be notified that the file he/she chose is not valid. This has to be done before clicking on upload. For example if he/she chooses file that is bigger than what is allowed in configuration, or some unsupported type, there won't be any validation error. Also this was put in place to prevent the user manipulating DOM through inspect element and allowing himself/herself upload of files bigger then fixed in configuration (or allowing any type of files...)